swan/Dockerfile

```diff
@@ -0,0 +1,192 @@
+    file \
+    gdbm-devel \
+    gettext \
+    jpackage-utils \
+    libXtst \
+    man \
+    passwd \
+    # pax \ -- Is not present in DNF repos.
+    perl-CGI \
+    perl-ExtUtils-MakeMaker \
+    perl-Test-Simple \
+    perl-devel \
+    rsyslog \
+    time \
+    xz-lzma-compat \
+    # Required by Geant4 (libXm)
+    openmotif \
+    # Required by Oracle
+    libaio \
+    # Required for key4hep
+    environment-modules 
+
+# Required by jupyter-resource-usage
+RUN pip install --no-deps --no-cache-dir psutil==5.8.0
+
+# Install all of our extensions
+# Ignore (almost all) dependencies because they have already been installed or come from CVMFS
+RUN pip install --no-deps --no-cache-dir \
+    dask-labextension==6.1.0 \
+    jupyter-resource-usage==0.6.0 \
+    hdfsbrowser==1.1.1 \
+    sparkconnector==2.4.6 \
+    sparkmonitor==2.1.1 \
+    swancontents==1.4.2 \
+    swanhelp==2.0.2 \
+    swanintro==1.0.0 \
+    swankernelenv==1.0.0 \
+    swannotebookviewer==1.2.0 \
+    swannotifications==1.0.0 \
+    swanoauthrenew==1.0.1 PyJWT \
+    swanshare==1.1.1 \
+    swanheader==1.0.0 \
+    swanportallocator==1.0.1
+            
+# swandask must be installed after its dependency dask-labextension to disable the server extension automatically
+RUN pip install --no-deps --no-cache-dir swandask==0.0.3
+
+# TODO: Review
+# Enable all the nbextensions and server extensions
+RUN jupyter nbextension install --py --system hdfsbrowser && \
+    jupyter nbextension install --py --system sparkconnector && \
+    jupyter nbextension install --py --system sparkmonitor && \
+    jupyter nbextension enable --py --system sparkmonitor && \
+    jupyter nbextension install --py --system swancontents && \
+    jupyter serverextension enable --py --system swancontents && \
+    jupyter nbextension install --py --system swanhelp && \
+    jupyter nbextension enable --py --system swanhelp && \
+    jupyter nbextension install --py --system swanintro && \
+    jupyter nbextension enable --py --system swanintro && \
+    jupyter serverextension enable --py --system swannotebookviewer && \
+    jupyter nbextension install --py --system swannotifications && \
+    jupyter nbextension enable --py --system swannotifications && \
+    jupyter serverextension enable --py --system swanoauthrenew && \
+    jupyter nbextension install --py --system swanshare && \
+    jupyter nbextension enable --py --system swanshare && \
+    jupyter serverextension enable --py --system swanshare && \
+    jupyter serverextension enable --py --system swanportallocator && \
+    # Force nbextension_configurator systemwide to prevent users disabling it
+    jupyter nbextensions_configurator enable --system && \
+    # Spark Monitor/Connector also need to be available to the user environment since they have kernel extensions
+    # PortAllocator is used by SparkConnector, so it needs to be available too
+    mkdir -p /usr/local/lib/swan/extensions && \
```

Comment:
This became too long, let's convert in a for loop that takes those extensions from an array.